### PR TITLE
Feat/content list

### PR DESCRIPTION
**Pull request by VladislavPokhyl**
Added followers  section  with infinite scroller  pagination.
Original example url : https://twitter.com/alexandrtovmach/followers
Access: appurl/followers
Related to tasks:
https://trello.com/c/vTcdxwzH/23-implement-user-item-component
https://trello.com/c/zxJAysAP/24-implement-user-list-container

image

### CODE REVIEW

File: src/components/FollowUserCard.tsx [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+  display: flex;
+  justify-content: space-between;
+`;
+
+const FollowerUserCard = (props: { user: MockUser }) => {
+  const user = props.user;
+  return (
+    <Card>
+      <Card.Body>
+        <HeaderSection>
```
Comment (DavidKorystov) [inline src/components/FollowUserCard.tsx]: u can use bootstrap styles here
```suggestion
        <section className="d-flex justify-content-between">
```

File: src/components/FollowUserCard.tsx [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+        </HeaderSection>
+
+        <Card.Title>{user.userLastName + user.userFirstName}</Card.Title>
+        <Card.Subtitle className="mb-2 text-muted">
+          @{user.userName}
+        </Card.Subtitle>
+
+        <Card.Text>{user.userCaption}</Card.Text>
+
+        <div style={{ marginTop: "10px" }}>
```
Comment (DavidKorystov) [inline src/components/FollowUserCard.tsx]:
```suggestion
        <div className="mt-2 ">
```

File: src/components/FollowUserDialogueButton.tsx [hunk 3]
```diff
@@ -0,0 +1,101 @@
...
+            onClick={props.onSubmit}
+          >
+            Unfollow
+          </Button>
+        </div>
+        <div>
+          <Button
+            className="rounded-pill"
+            variant="outline-dark"
+            style={{ width: "100%", marginTop: "20px" }}
```
Comment (DavidKorystov) [inline src/components/FollowUserDialogueButton.tsx:40]:
```suggestion
            className="w-100 mt-3"
```

File: src/components/FollowUserDialogueButton.tsx [hunk 4]
```diff
@@ -0,0 +1,101 @@
...
+        <p>
+          Their Tweets will no longer show up in your home timeline. You can
+          still view their profile, unless their Tweets are protected.
+        </p>
+
+        <div>
+          <Button
+            className="rounded-pill"
+            variant="dark"
+            style={{ width: "100%", marginTop: "20px" }}
```
Comment (DavidKorystov) [inline src/components/FollowUserDialogueButton.tsx:30]: same

File: src/components/FollowUserItem.tsx [hunk 5]
```diff
@@ -0,0 +1,93 @@
...
+        </MiddleSectionWrapper>
+        <RightSectionWrapper>
+          <FollowButton
+            followStatus={user.followStatus}
+            followerName={user.userName}
+          />
+        </RightSectionWrapper>
+      </UserItemHeader>
+
+      <UserFollowerItemContent>
```
Comment (DavidKorystov) [inline src/components/FollowUserItem.tsx:69]:
```suggestion
      <p className="d-flex h-75">
```

File: src/components/FollowUserItem.tsx [hunk 6]
```diff
@@ -0,0 +1,93 @@
...
+const ListItemWoFooter = (props: { user: MockUser }) => {
+  const user = props.user;
+  const popover = (
+    <Popover>
+      <FollowCard user={user} />
+    </Popover>
+  );
+  return (
+    <LiWrapper>
+      <UserItemHeader>
```
Comment (DavidKorystov) [inline src/components/FollowUserItem.tsx:37]:
```suggestion
      <div className="h-25 d-flex">
```

File: src/containers/InfiniteList.tsx [hunk 7]
```diff
@@ -0,0 +1,26 @@
+import InfiniteScroll from "react-infinite-scroller";
+
+import { PaginationParamsResult } from "../types/mock-api-types";
+export interface InfiniteScrollerListProps {
+  // eslint-disable-next-line
+  showMore(page: number): any;
+  data: PaginationParamsResult;
+  // eslint-disable-next-line
+  itemComponent: (item: any) => JSX.Element;
```
Comment (KirillSerg) [inline src/containers/InfiniteList.tsx:9]: tell me pleas why you use '=>'? What is this?

File: src/pages/Followers.tsx [hunk 8]
```diff
@@ -0,0 +1,72 @@
...
+        userName: "user name" + suffix,
+        userUrl: "qweqwe",
+        userPhotoUrl:
+          "https://encrypted-tbn0.gstatic.com/images?q=tbn:ANd9GcRyF9H3AGaJdh8zjSZlc4yX3ZSdDGw-kijLQQ&usqp=CAU",
+        userCaption:
+          "Lorem Ipsum is simply dummy text of the printing and typesetting industry. Lorem Ipsum has been the industry's standard dummy text ever since the 1500s, when an unknown printer took a galley of type and scrambled it to make a type specimen book.  ",
+        followers: Math.floor(Math.random() * 1000),
+        following: Math.floor(Math.random() * 1000),
+      });
+      console.log(id);
```
Comment (KirillSerg) [inline src/pages/Followers.tsx]: we need this and next?

**Review by KirillSerg — APPROVED** [review]

File: src/containers/InfiniteList.tsx (hunk 7 above)
Comment (VladislavPokhyl) [reply src/containers/InfiniteList.tsx:9]: Not sure if I can explain this better then official docs.  Basically in this example I specified return type of the function.
this function will match the same
```
function func(item : any):JSX.Element  {
//...
const result : JSX.Element = *some JSX.element variable*
return result
}
```
https://www.typescriptlang.org/docs/handbook/functions.html#this-and-arrow-functions

File: src/pages/Followers.tsx (hunk 8 above)
Comment (VladislavPokhyl) [reply src/pages/Followers.tsx]: Not  sure if I get the question , but if you are asking about  line #41 , then I just forgot to remove it after debugging. Will remove it with my next PR (unless it is urgent to remove this kind of misstypes)

File: src/App.tsx [hunk 9]
```diff
@@ -6,6 +6,7 @@ import AboutPage from "./pages/About";
 import AuthorizationPage from "./pages/Authorization";
 import BookmarksPage from "./pages/Bookmarks";
 import ExplorePage from "./pages/Explore";
+import RecomendedFollowsPage from "./pages/Followers";
```
Comment (alexandrtovmach) [inline src/App.tsx]:
```suggestion
import FollowersPage from "./pages/Followers";
```

File: src/components/FollowUserCard.tsx [hunk 10]
```diff
@@ -0,0 +1,49 @@
+import { Card, Image } from "react-bootstrap";
```
Comment (alexandrtovmach) [inline src/components/FollowUserCard.tsx]: UserCard.tsx

File: src/components/FollowUserCard.tsx [hunk 11]
```diff
@@ -0,0 +1,49 @@
...
+
+import { MockUser } from "../types/mock-api-types";
+import FollowButton from "./FollowUserDialogueButton";
+
+const HeaderSection = styled.div`
+  display: flex;
+  justify-content: space-between;
+`;
+
+const FollowerUserCard = (props: { user: MockUser }) => {
```
Comment (alexandrtovmach) [inline src/components/FollowUserCard.tsx]:
```suggestion
const FollowerUserCard = ({ user }: { user: MockUser }) => {
```

File: src/components/FollowUserItem.tsx [hunk 12]
```diff
@@ -0,0 +1,93 @@
+import {
```
Comment (alexandrtovmach) [inline src/components/FollowUserItem.tsx]: UserListItem.tsx

File: src/pages/Followers.tsx (hunk 8 above)
Comment (alexandrtovmach) [reply src/pages/Followers.tsx]: remove console.log

File: src/pages/Followers.tsx [hunk 13]
```diff
@@ -0,0 +1,72 @@
...
+    }
+    console.log(mockData);
```
Comment (alexandrtovmach) [inline src/pages/Followers.tsx]: remove console.log

**Review by alexandrtovmach — CHANGES_REQUESTED** [review]
few changes to do before merge

**Review by alexandrtovmach — APPROVED** [review]

**Comment by alexandrtovmach** [comment]
@VladislavPokhyl all requested changes above please add with another PR

**Pull request merged**